src/linear_algebra/general_linear_group.lean

```diff
@@ -71,3 +71,29 @@ rfl
 end general_linear_group
 
 end linear_map
+
+section
+
+variables {M R : Type*} [ring R] [add_comm_group M] [module R M] (T : M →ₗ[R] M)
+
+/-- any invertible linear map can be written as a linear equivalence -/
+def linear_equiv.of_invertible [invertible T] : M ≃ₗ[R] M :=
+linear_map.general_linear_group.to_linear_equiv (unit_of_invertible T)
+
+lemma linear_equiv.coe_of_invertible [invertible T] :
+  ⇑(linear_equiv.of_invertible T) = T := rfl
+
+lemma linear_equiv.coe_linear_map_of_invertible [invertible T] :
+  ↑(linear_equiv.of_invertible T) = T := by { ext, refl }
+
+@[simp] lemma linear_map.of_invertible_symm_eq_inv_of [invertible T] :
+  ↑((linear_equiv.of_invertible T).symm) = (⅟ T) := by { ext, refl }
```

Comment:
Can you wrap this in a `linear_equiv` namespace? Also, is the last lemma deliverately in the `linear_map` namespace?

Reply:
Yeah, I think the last lemma was meant to be in the `linear_map` namespace. Do you think I should change it?